internal/syncd/integrations/k8s/informer.go

```diff
@@ -0,0 +1,131 @@
+package k8s
+
+import (
+	"context"
+	"fmt"
+
+	shipitv1beta1 "ship-it-operator/api/v1beta1"
+	"ship-it/internal/unstructured"
+
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	toolscache "k8s.io/client-go/tools/cache"
+	"sigs.k8s.io/controller-runtime/pkg/cache"
+)
+
+const imageRepositoriesIndex = "ImageRepositoriesIndex"
+
+var errCacheSyncFailed = errors.New("repository informer: image repository cache sync failed")
+
+func errUnexpectedObjectType(obj interface{}) error {
+	return fmt.Errorf("repository informer: unexpected object type %T", obj)
+}
+
+func eventHandler(indexer toolscache.Indexer) *toolscache.ResourceEventHandlerFuncs {
+	// TODO: should we log indexer errors?
```

Comment:
thoughts on this? the `indexer.{Add,Delete}` methods return an error, but it's not really actionable. Do we care enough to log it?

Reply:
It may be good to know that it was these methods that failed when debugging. However, considering the indexer will be called upon often, it could clutter the logs.